lib/splitclient-rb/clients/split_client.rb

```diff
@@ -245,25 +247,54 @@ def treatments(key, split_names, attributes = {}, calling_method = 'get_treatmen
       bucketing_key = bucketing_key ? bucketing_key.to_s : nil
       matching_key = matching_key ? matching_key.to_s : nil
 
-      evaluator = Engine::Parser::Evaluator.new(@segments_repository, @splits_repository, @config, true)
-      start = Time.now
-      impressions = []
-      treatments_labels_change_numbers =
-        @splits_repository.splits(sanitized_split_names).each_with_object({}) do |(name, data), memo|
-          memo.merge!(name => treatment(key, name, attributes, data, false, true, evaluator, calling_method, impressions))
-        end
+      if !@config.split_validator.valid_get_treatments_parameters(calling_method, key, sanitized_feature_flag_names, matching_key, bucketing_key, attributes)
+        to_return = Hash.new
+        sanitized_feature_flag_names.each {|name|
+          to_return[name.to_sym] = control_treatment_with_config
+        }
+        return to_return
+      end
 
-      record_latency(calling_method, start)
-      @impressions_manager.track(impressions)
+      if !ready?
+        impressions = []
+        to_return = Hash.new
+        sanitized_feature_flag_names.each {|name|
+          to_return[name.to_sym] = control_treatment_with_config
+          impressions << @impressions_manager.build_impression(matching_key, bucketing_key, name.to_sym, control_treatment_with_config.merge({ label: Engine::Models::Label::NOT_READY }), { attributes: attributes, time: nil })
+        }
+        @impressions_manager.track(impressions)
+        return to_return
+      end
 
-      split_names_keys = treatments_labels_change_numbers.keys
-      treatments = treatments_labels_change_numbers.values.map do |v|
+      valid_feature_flag_names = []
+      sanitized_feature_flag_names.each { |feature_flag_name|
+        valid_feature_flag_names << feature_flag_name unless feature_flag_name.nil?
+      }
+      start = Time.now
+      impressions_total = []
+
+      feature_flags = @splits_repository.splits(feature_flag_names)
```

Comment:
why are you `splits_repository` here? also you should use `Evaluator.evaluate_feature_flags` in this method

Reply:
I did in the beginning, but it became too complicated to cover all scenarios of result treatment, it was cleaner to fetch splits in client then evaluate each single split.